solr/test-framework/src/java/org/apache/solr/util/RandomizeSSL.java

```diff
@@ -104,10 +105,10 @@ public SSLRandomizer(double ssl, double clientAuth, String debug) {
     public SSLTestConfig createSSLTestConfig() {
       // even if we know SSL is disabled, always consume the same amount of randomness
       // that way all other test behavior should be consistent even if a user adds/removes @SuppressSSL
-      
-      final boolean useSSL = TestUtil.nextInt(LuceneTestCase.random(), 0, 999) <
+      Random random = new Random();
```

Comment:
Because we use randoms differently in the benchmark stuff and don't want to be stuck with Randomized testing randoms.

This change was not intended to stick here though, that was from early workarounds.

I don't really want the random enforcement / support from the test framework for a couple reasons, but this is simply something not removed - the problem being if you used the mincluster and jetty with jetty.testMode=true and did not launch things via carrot randomized runner, it will *sometimes* detect you dont have a test randomized context for a thread and fail you - but we are not using randomized runner or junit. Currently, I work around needing this workaround by not using the jetty.testMode sys prop path and adding another sys prop hack atm for where starting the mini cluster is a bit too tied into the test framework and carrot random reqs.

Java 7 and up has essentially made Random obsolete, so there needs to be some separate regardless, because we don't use carrot2 junit runners for benchmarks, but it's also much preferable, faster, improved to avoid Random entirely and use ThreadLocalRandom and SplittableRandom instead, so i try and use them in the benchmarks.

Reply:
@dweiss any thoughts on the above?

In general, it'd be nice to have code that uses a consistent random number and be compatible with RandomizedTesting yet not actually depend on RnadomizedRunner (e.g. because the code _sometimes_ runs without RR -- e.g. benchmark).

Reply:
My thought is that if you use independently initialized randoms then the entire concept of randomized testing fails - you won't be able to reproduce your issues. Random isn't really gone. ThreadLocalRandom is a Random. SplittableRandom isn't, but you could make a delegate if needed. 

Also, as I many times pointed out, if performance is the key, use any random implementation you wish but start from the framework's-provided initial seed. Then you can hope for reproducible behavior but at the same time have a faster random seed generator where it's needed.

Reply:
Yes, in both a test context or benchmark context, we want reproducibility.  Thus `new Random()` (that which I commented on here) is never okay.  Maybe Solr requires some new random supplier utility that either uses RandomizedTesting if it's available (detect via reflection?), and if not then something vaguely similar with a fixed seed (possibly with sys prop) & ThreadLocalRandom?

Reply:
Anything, really. If it's not running under the junit (randomized) test runner then you won't be able to use any of the randomized context stuff - it'll throw illegal state exceptions. But you can try to pass/ reuse the system property tests.seed to initialize that "local" random without entering randomizedtesting code. This is done in StringHelper, for example:
```
  static {
    String prop = System.getProperty("tests.seed");
    if (prop != null) {
      // So if there is a test failure that relied on hash
      // order, we remain reproducible based on the test seed:
      GOOD_FAST_HASH_SEED = prop.hashCode();
    } else {
      GOOD_FAST_HASH_SEED = (int) System.currentTimeMillis();
    }
  }
```